pkg/issuer/acme/http/solver/solver.go

```diff
@@ -55,18 +55,36 @@ func (h *HTTP01Solver) Listen(log logr.Logger) error {
 		"listen_port", h.ListenPort,
 	)
 
-	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	h.Server = http.Server{
+		Addr:              fmt.Sprintf(":%d", h.ListenPort),
+		Handler:           h.challengeHandler(log),
+		ReadHeaderTimeout: defaultReadHeaderTimeout, // Mitigation for G112: Potential slowloris attack
+	}
+
+	return h.Server.ListenAndServe()
+}
+
+func (h *HTTP01Solver) challengeHandler(log logr.Logger) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
 		// extract vars from the request
-		host := strings.Split(r.Host, ":")[0]
+		var hosts = make(map[string]string, 2)
+		hosts[r.Host] = ""
+		parts := strings.Split(r.Host, ":")
+		if len(parts) > 1 {
+			hosts[strings.Join(parts[:len(parts)-1], ":")] = ""
+		}
+
```

Comment:
suggestion: Would https://pkg.go.dev/net/netip#ParseAddrPort by useful for this? It seems to do what we want without us having to add custom parsing?